spec/index.adoc

```diff
@@ -3454,6 +3479,67 @@ S with
       }
 ....
 
+=== IC Management Canister: Code eviction
+
+Only the controllers of the given canister can install new code.
+This method changes the code of an _existing_ canister, preserving its stable memory.
+This involves invoking the `canister_pre_upgrade` method on the old and `canister_post_upgrade` method on the new canister, which must succeed and must not invoke other methods.
+
+Conditions::
+....
+    S.messages = Older_messages · CallMessage M · Younger_messages
+    (M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+    M.callee = ic_principal
+    M.method_name = 'install_code'
+    M.arg = candid(A)
+    Mod = parse_wasm_mod(A.wasm_module)
+    Public_custom_sections = parse_public_custom_sections(A.wasm_module)
+    Private_custom_sections = parse_private_custom_sections(A.wasm_module)
+    A.mode = evict
```

Comment:
This mode should also be introduced in the type definition in the section "The system state"

Reply:
Nevermind; this comment affects the canister history MR after this one is merged.